pkg/rancher-desktop/pages/marketplace/details.vue

```diff
@@ -0,0 +1,352 @@
+<template>
+  <div class="extension">
+    <nuxt-child />
+    <div class="extension-content">
+      <div>
+        <Banner v-if="error" color="error" class="banner">
+          {{ error }}
+        </Banner>
+        <Banner v-if="response" color="info" class="banner">
+          {{ response }}
+        </Banner>
+        <div v-html="htmlContent" />
+
+        <img :src="getPreviewImage" alt="" />
+      </div>
+      <div class="extension-content-sidebar">
+        <button
+          v-if="!isInstalled"
+          class="btn btn-xs role-primary"
+          @click="appInstallation()"
+        >
+          <slot v-if="loading" name="loading" :is-loading="loading">
+            <loading-indicator>
+              {{ t('marketplace.sidebar.installButton.loading') }}
+            </loading-indicator>
+          </slot>
+
+          <span v-if="!loading">
+            {{ t('marketplace.sidebar.installButton.label') }}</span>
+        </button>
+
+        <button
+          v-else
+          class="btn btn-xs role-primary"
+          @click="appInstallation('uninstall')"
+        >
+          <slot v-if="loading" name="loading" :is-loading="loading">
```

Comment:
I recommend that we remove the slots if they're not needed. If they are, we could redesign these buttons so that we only expose a single slot.